pydantic/functional_serializers.py

```diff
@@ -181,7 +181,7 @@ def model_serializer(
     *,
     mode: Literal['plain', 'wrap'] = 'plain',
     json_return_type: _core_schema.JsonReturnTypes | None = None,
-) -> Callable[[Any], _decorators.PydanticDescriptorProxy[Any]] | _decorators.PydanticDescriptorProxy[Any]:
+) -> Callable[[Any], Any]:
```

Comment:
Is it a problem that the non-callable return type was removed here? I mean, I'm guessing it's not a problem, but it would it be more careful in some way? Happy to leave like this if preferable just want to make sure it wasn't an oversight.

Reply:
I just pushed a slightly better version. We can get even more granular (e.g. defining a couple of `Protocol`s like we do form some of the other validators/serializers) but I don't want to do that in this PR, I'm only changing it because it was causing issues with the test I'm adding here.

Reply:
pls can we add (if we don't already have) an example of both `@model_validator` and `@model_validator(...)` in mypy tests please.